Add websocket reconnect

tbone1020 opened the pull request.

efjacobson commented:
note: we need to get final sign off from Sam on the wording used here

efjacobson commented:
@tbone1020 @arunkarnati final wording is this:

> Your active edits connection is unstable; we have attempted to reconnect but it is beyond repair. Other users may not be able to see your activity. Please save any pending work and refresh the page. If this problem persists you may want to alert team members of your activity. Thank you.

with this change, there is no longer a need to attempt to reconnect

arunkarnati commented on CHANGELOG.md:
```diff
@@ -1,6 +1,9 @@
 
 
+## N.N.N
+* issue #1225: Look For Solutions To Reconnect CMS Web Socket When It Disconnects.
```
@tbone1020 - please use an issue number if it is referring to a ticket created in this repo. e.g., https://github.com/triniti/cms-js/issues/232

arunkarnati reviewed (changes requested):
@tbone1020 - according to Eric's comment here, https://github.com/triniti/cms-js/pull/255#issuecomment-702912409, we don't need to make any additional attempts to reconnect. Please remove the wants-to-reconnect code and display an alert message with the wording quoted in the above comment.

arunkarnati commented:
> @tbone1020 - according to Eric's comment here, [#255 (comment)](https://github.com/triniti/cms-js/pull/255#issuecomment-702912409), we don't need to make any additional attempts to reconnect. Please remove the wants-to-reconnect code and display an alert message with the wording quoted in the above comment.

@tbone1020 - checking in to see if you had a chance to look at my feedback and have any questions?

efjacobson commented:
@arunkarnati @tbone1020 they are actually still waiting on final language here, lets park this until that is settled

arunkarnati commented:
> @arunkarnati @tbone1020 they are actually still waiting on final language here, lets park this until that is settled

10-4

efjacobson commented on src/plugins/raven/sagas/connectionFlow.js:
```diff
@@ -61,6 +64,23 @@ export default function* (raven) {
+        },
+      });
+
+      if (userWantsToReconnect.value && userWantsToReconnect.value === true) {
```
@tbone1020 you can remove all this reconnect stuff since they dont want it anymore - it's just a notification now

efjacobson commented on src/plugins/raven/sagas/connectionFlow.js:
```diff
@@ -64,23 +62,6 @@ export default function* (raven) {
     attempts += 1;
     if (attempts > 10) {
       console.error('raven::connectionFlow::exceeded_10_attempts'); // eslint-disable-line no-console
-      const userWantsToReconnect = yield swal.fire({
```
@tbone1020 you do still want to fire the swal, you just dont need to do anything else

efjacobson commented on src/plugins/raven/sagas/connectionFlow.js:
```diff
@@ -17,6 +17,7 @@ const maxConnectionAttempts = 5;
 function* connect(raven) {
+  const maxConnectionAttempts = 5;
   let error;
   for (let i = 0; i < maxConnectionAttempts; i += 1) {
```
@tbone1020 you can just hardcode this this to 5 like it was before since we wont be changing it

tbone1020 commented:
@efjacobson I added sweet alert back.

efjacobson commented:
@tbone1020 also, it looks like the `warning` type is the closest to the mock https://sweetalert2.github.io/#icons

tbone1020 commented:
@efjacobson hard coded maxConnectionAttemps, changed alert to "warning", and removed "yield" from in front of the sweet alert since we're not waiting for users response.

efjacobson commented on src/plugins/raven/sagas/connectionFlow.js:
```diff
@@ -40,7 +41,7 @@ function* connect(raven) {
 export default function* (raven) {
   const connectsChannel = yield actionChannel(actionTypes.CONNECTION_OPENED, buffers.dropping(1));
-  let attempts = 0;
+  let attempts = 10;
```
@tbone1020 leave this at 0 where it was, otherwise it will only attempt a single time

tbone1020 commented:
@efjacobson changed it to 10 for testing purposes. It has been changed back to 0.
